src/JSPool.Tests/JsPoolTests.cs

```diff
@@ -21,7 +21,7 @@ public class JsPoolTests
 		public void ConstructorCreatesEngines()
 		{
 			var factory = new Mock<IEngineFactoryForMock>();
-			factory.Setup(x => x.EngineFactory()).Returns(new Mock<IJsEngine>().Object);
+			factory.Setup(x => x.EngineFactory()).Returns(() => new Mock<IJsEngine>().Object);
```

Comment:
The metadata is keyed on the engine. The factory returns the same engine all the time. It works previously due to maintaining a separate engineCount, which did not take into account the uniqueness of the engine. It failed with change to use metadata.Count